actors/builtin/miner/miner_state.go

```diff
@@ -748,49 +801,42 @@ func (st *State) LoadSectorInfos(store adt.Store, sectors *abi.BitField) ([]*Sec
 // Loads info for a set of sectors to be proven.
 // If any of the sectors are declared faulty and not to be recovered, info for the first non-faulty sector is substituted instead.
 // If any of the sectors are declared recovered, they are returned from this method.
-func (st *State) LoadSectorInfosForProof(store adt.Store, provenSectors *abi.BitField) (sectorInfos []*SectorOnChainInfo, recoveries *bitfield.BitField, err error) {
-	// Extract a fault set relevant to the sectors being submitted, for expansion into a map.
-	declaredFaults, err := bitfield.IntersectBitField(provenSectors, st.Faults)
-	if err != nil {
-		return nil, nil, xerrors.Errorf("failed to intersect proof sectors with faults: %w", err)
-	}
-
-	recoveries, err = bitfield.IntersectBitField(declaredFaults, st.Recoveries)
+func (st *State) LoadSectorInfosForProof(store adt.Store, provenSectors, expectedFaults []*abi.BitField) ([]*SectorOnChainInfo, error) {
```

Comment:
Any reason not to merge the bitfields outside fo this function?

Reply:
No, either way is fine.